### ISSUE TITLE
Crash while opening a librairy

### ISSUE THREAD

**Issue by SamuelMorel** [files-community/Files#7037]
### Description

When I open Files on two specific library 
* Home -> one underlying folder = C:\Users\<myUser>
* Images -> one underlying folder = C:\Users\<myUser>\images

### Steps To Reproduce

Just open Files in "images" Library

### Expected behavior

Files doesn't crash.

### Files Version

Version : 2.0.34.0 OS Version: 10.0.18363.1916

### Windows Version

Windows 10 enterprise 1909 build 18363.1916

### Relevant Assets

```
2021-11-23 14:14:45.6852|WARN|CheckForUpdatesAsync|Could not fetch updates.
System.ArgumentOutOfRangeException: L’opération a tenté d’accéder aux données en dehors de la plage active (Exception from HRESULT: 0x8000000B)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at Files.Helpers.AppUpdater.<CheckForUpdatesAsync>d__2.MoveNext() + 0x268
```

**Comment by gave92** [comment]
Unfortunately that's only a warning. Could you post the full log files?
Is the crash repeatable?
When it crashes, do you get a notification asking for you to report the bug?

**Comment by SamuelMorel** [comment]
Thanks for your response.

The crash can be repeated every time I reproduce the given steps -> open one of the two libraries

No notification triggered, juste the main window that closes.

About the logs, here they are, but unfortunately there is nothing more in the logs...
[debug.log](https://github.com/files-community/Files/files/7589543/debug.log)
[debug_fulltrust.log](https://github.com/files-community/Files/files/7589554/debug_fulltrust.log)

.

**Comment by gave92** [comment]
Thanks for the info, would you check if the crash happens also if you do the following?
- In Files settings > Files and folders, turn off "Enable individual preferences for folders"
- Navigate to any directory (that does not crash) and set e.g Tiles layout mode
- Navigate to one of the libraries

**Comment by SamuelMorel** [comment]
You're right, by setting this parameter, is works like a charm ! 

Thanks 👍

**Comment by gave92** [comment]
Glad it works, unfortunately it will use Tiles view mode in all folders :)

Just for confirmation would you try to set the layout to Details and check if the issue reappears depending on "Enable individual preferences for folders" being on or off?

**Comment by SamuelMorel** [comment]
Actually is does not, detailed view is still used.

I had a wired behaviour, after disable the setting, all went well (in detailed view) and after that, no matter if I enable or disable the setting, it still works just fine ! 
I've try it another time by : 
* enable the setting
* create a library 
* try to display the librairy -> crash
* disable the setting
* try to display the librairy -> ok

I have to say that I don't really understand...

The tile view mode does not seems to change something.

**Comment by SamuelMorel** [comment]
> 

Oh no I'm sorry, I did not refreshed my view.
My previous message is wrong, I'll let you know my test result tomorrow. 
Thanks for your help.

**Comment by gave92** [comment]
No rush, thanks for all the testing

**Comment by SamuelMorel** [comment]
Hi @gave92,

Unfortunately I did not reproduce the behaviour after a switch of the parameter... 
Maybe the restart of the system did change something, or the switch got it fix somehow.

Anyway, I do not have the bug anymore, thanks for you help !

**Comment by warappa** [comment]
I roughly followed the steps to do the `Enable individual preferences for folders`-dance, but I
* Being on start screen
* disabled the setting
* navigated to "Pictures" - and this time it did not crash
* navigated to C-drive
* enabled the setting
* navigated to "Pictures" - and this time it did also not crash 🎉

But:
If I navigate to another library, it will crash again. After **this** crash, navigating to "Pictures" crashes again.

It seems to me, that after a library-folder is accessed for the first time, it is _cached_, so the `Enable individual preferences for folders`-bug does not trigger.  
This could mean changing this setting only applies to not previously accessed folders.

**Comment by gave92** [comment]
Ok my understanding is that it only crashes when a folder is in details view & grouped & system language is not english. That issue has been fixed in #7202.
